lms/djangoapps/bulk_email/tests/test_course_optout.py

```diff
@@ -87,6 +88,52 @@ def test_optout_course(self):
         self.assertEqual(len(mail.outbox), 1)
         self.assertEqual(mail.outbox[0].to[0], self.instructor.email)
 
+    def test_optout_using_unsubscribe_link_in_email(self):
+        """
+        Make sure email (with Unicode characters) send to all goes there.
+        """
+        self.client.logout()
+
+        self.client.login(username=self.instructor.username, password="test")
+        self.navigate_to_email_view()
+
+        test_email = {
+            'action': 'Send email',
+            'send_to': '["learners"]',
+            'subject': 'Checking unsubscribe link in email',
+            'message': 'test message for all'
+        }
+        response = self.client.post(self.send_mail_url, test_email)
+        self.assertEqual(json.loads(response.content.decode('utf-8')), self.success_content)
+
+        # check unsubscribe link in template
+        plain_template = mail.outbox[0].body
+        html_template = mail.outbox[0].alternatives[0][0]
+
+        assert u'bulk_email/email/optout/' in plain_template
+        assert u'bulk_email/email/optout/' in html_template
+
+        unsubscribe_link = re.findall(r"""/bulk_email/email.*(?='>)""", mail.outbox[0].alternatives[0][0])[0].strip()
+
+        response = self.client.get(unsubscribe_link)
+
+        self.assertContains(response, 'unsubscribe')
```

Comment:
Remove this redundant code, it's already been tested here https://github.com/edx/edx-platform/pull/22585/files#diff-1966eaf949ad6e452c5e28e161b9d8a4R607